src/widgets/powered-by/powered-by.tsx

```diff
@@ -29,36 +37,48 @@ const renderer = ({ containerNode, cssClasses }) => (
   }
 };
 
-/**
- * @typedef {Object} PoweredByWidgetCssClasses
- * @property  {string|string[]} [root] CSS classes added to the root element of the widget.
- * @property  {string|string[]} [link] CSS class to add to the link.
- * @property  {string|string[]} [logo] CSS class to add to the SVG logo.
- */
-
-/**
- * @typedef {Object} PoweredByWidgetParams
- * @property {string|HTMLElement} container Place where to insert the widget in your webpage.
- * @property {string} [theme] The theme of the logo ("light" or "dark").
- * @property {PoweredByWidgetCssClasses} [cssClasses] CSS classes to add.
- */
-
-/**
- * The `poweredBy` widget is used to display the logo to redirect to Algolia.
- * @type {WidgetFactory}
- * @devNovel PoweredBy
- * @category metadata
- * @param {PoweredByWidgetParams} widgetParams PoweredBy widget options. Some keys are mandatory: `container`,
- * @return {Widget} A new poweredBy widget instance
- * @example
- * search.addWidgets([
- *   instantsearch.widgets.poweredBy({
- *     container: '#poweredBy-container',
- *     theme: 'dark',
- *   })
- * ]);
- */
-export default function poweredBy(widgetParams) {
+export type PoweredByCSSClasses = {
```

Comment:
We can get the required form in the component and do this here:

```ts
type PoweredByCSSClasses = Partial<PoweredByComponentCSSClasses>
```

Reply:
or even required here, and partial https://github.com/algolia/instantsearch.js/pull/4756/files#diff-a1c4756cb305947c3149d511a59b64ba6bd6f268eb5eed4f036eaa95d76d4e19R72

Reply:
Does this make sense to you? https://github.com/algolia/instantsearch.js/pull/4756/commits/0b93a236f1c4c693a3f085aadb0acf04fedf5090